README.md
# AdiBags_BfAZoneItems #

**This is a fork of the [Add-on](https://www.curseforge.com/wow/addons/adibags-zone-items "AdiBags - BFA Zone Items on CurseForge") by Eek! - Earthen Ring**

- - - -

I have added additional items to the list of sorting that was not in the base add-on. 

- I have added items used to create Essences to be sorted into the Essence group.
- I have added sorting out Black Empire BoA tokens. 
- I have added in reputation items from all patches to be sorted into Reputation group. 

CHANGES.txt
06.26.19    v 1.0 Functional first pass
06.27.19    v 1.01 Added additional items 
07.01.19    v 1.02 Added additional items 
07.07.19    v 1.03 Added additional items 
07.10.19    v 1.04 Split data into second file, added items, and added filters for Benthic and Rep items
08.06.19    v 1.05 Added filters for Heart of Azeroth Essences, and prioritizing item groups by current zone
02.06.20    v 1.06 Added filters for Patch 8.3 items

ZoneItemData.lua
--[[
Datafile for AdiBags_BfAZoneItems
]]
local addonName, addon = ...
local L = addon.L


-- Zone Item Data
addon.ZONE_ITEMS = {
  "163217^BfA^Reputation^Azeroth's Tear",
  "174502^BfA^Reputation^Tear of Azeroth", 
  "166501^BfA^Reputation^Soggy Page",
  "174519^BfA^Reputation^Verdant Hills of Chokingvine - Page 17",
  "163620^BfA^Reputation^Island Flotsam",
  "174518^BfA^Reputation^Jani Figurine",
  "163619^BfA^Reputation^Golden Beetle", 
  "174506^BfA^Reputation^Golden Insect Wings",
  "163618^BfA^Reputation^Shimmering Shell",
  "174501^BfA^Reputation^Ornate Voldunai Jewelry",
  "163616^BfA^Reputation^Dented Coin", 
  "174504^BfA^Reputation^Proudmoore War Copper", 
  "163614^BfA^Reputation^Exotic Spices", 
  "174503^BfA^Reputation^Exotically Spiced Carrots",
  "163615^BfA^Reputation^Lost Sea Scroll", 
  "174505^BfA^Reputation^Tide-Speaker's Tome", 
  "163617^BfA^Reputation^Rusted Alliance Insignia", 
  "53117^BfA^Reputation^Rusted Alliance Insignia", 
  "163621^BfA^Reputation^Rusted Horde Insignia", 
  "174508^BfA^Reputation^Fallen Soldier's Insignia", 
  "174522^BfA^Reputation^Waveswept Abyssal Conch", 
  "174523^BfA^Reputation^Waveswept Abyssal Conch",
  "170081^BfA^Reputation^Abyssal Conch", 
  "170079^BfA^Reputation^Abyssal Conch", 
  "173947^BfA^Reputation^Glittering Abyssal Conch", 
  "173948^BfA^Reputation^Glittering Abyssal Conch", 
  "169941^BfA^Reputation^Ceremonial Ankoan Scabbard", 
  "169942^BfA^Reputation^Vibrant Sea Blossom", 
  "168497^BfA^Reputation^Rustbolt Resistance Insignia", 
  "167661^BfA^Reputation^Energized Lightning Cod", 
  "167662^BfA^Reputation^Solarsprocket Barbel", 
  "167655^BfA^Reputation^Bolted Steelhead", 
  "167663^BfA^Reputation^Tasty Steelfin", 
  "167654^BfA^Reputation^Bottom Feeding Stinkfish", 
  "167656^BfA^Reputation^Pond Hopping Springfish", 
  "167657^BfA^Reputation^Shadowy Cave Eel", 
  "167660^BfA^Reputation^Sludge-fouled Carp", 
  "167659^BfA^Reputation^Spitting Clownfish", 
  "167658^BfA^Reputation^Mechanical Blowfish", 
  "169478^BfA^Benthic^Benthic Bracers", 
  "169480^BfA^Benthic^Benthic Chestguard", 
  "169481^BfA^Benthic^Benthic Cloak", 
  "169477^BfA^Benthic^Benthic Girdle", 
  "169485^BfA^Benthic^Benthic Guantlets", 
  "169479^BfA^Benthic^Benthic Helm", 
  "169482^BfA^Benthic^Benthic Leggings", 
  "169484^BfA^Benthic^Benthic Spaulders", 
  "169483^BfA^Benthic^Benthic Treads", 
  "169774^BfA^Essence^Progression Sprocket", 
  "174279^BfA^Essence^Encrypted Ny'alothan Text", 
  "174287^BfA^Essence^Sparks of Unwavering Strength", 
  "174288^BfA^Essence^Breath of Everlasting Spirit",
  "169590^BfA^Essence^Burgeoning Battlefield Furor",  
  "169694^BfA^Essence^Aqueous Reliquary", 
  "169491^BfA^Essence^Focused Life Anima", 
  "169868^Mechagon^^Anti-Gravity Pack", 
  "167062^Mechagon^^Armored Vaultbot Key", 
  "168233^Mechagon^^Bawld-371", 
  "168045^Mechagon^^Beastbot Power Pack", 
  "168327^Mechagon^^Chain Ignitercoil", 
  "166972^Mechagon^^Emergency Powerpack", 
  "166973^Mechagon^^Emergency Repair Kit", 
  "166971^Mechagon^^Empty Energy Cell", 
  "166970^Mechagon^^Energy Cell", 
  "168961^Mechagon^^Exothermic Evaporator Coil", 
  "168832^Mechagon^^Galvanic Oscillator", 
  "168217^Mechagon^^Hardened Spring", 
  "168952^Mechagon^^Hardened Spring", 
  "167649^Mechagon^^Hundred-Fathom Lure", 
  "167562^Mechagon^^Ionized Minnow", 
  "169872^Mechagon^^Irontide Lockbox Key", 
  "169878^Mechagon^^Irradiated Undercoat", 
  "168215^Mechagon^^Machined Gear Assembly", 
  "168950^Mechagon^^Machined Gear Assembly", 
  "169873^Mechagon^^Mechanized Supply Key", 
  "167071^Mechagon^^Mechano-Treat", 
  "169218^Mechagon^^Old Rusty Key", 
  "169675^Mechagon^^Orange Paint Filled Bladder", 
  "169114^Mechagon^^Personal Time Displacer", 
  "169470^Mechagon^^Pressure Relief Valve", 
  "169610^Mechagon^^S.P.A.R.E. Crate", 
  "168262^Mechagon^^Sentry Fish", 
  "166846^Mechagon^^Spare Parts", 
  "168216^Mechagon^^Tempered Plating", 
  "168951^Mechagon^^Tempered Plating", 
  "168213^Mechagon^^Tensile Driveshaft", 
  "167075^Mechagon^^Ultrasafe Transporter: Mechagon", 
  "170186^Nazjatar^^Abyss Pearl", 
  "170079^Nazjatar^Reputation^Abyssal Conch", 
  "170184^Nazjatar^Reputation^Ancient Reefwalker Bark", 
  "167910^Nazjatar^^Bag of Who-Knows-What", 
  "169782^Nazjatar^^Beckoner's Rosetta Stone", 
  "170189^Nazjatar^^Blind Eye", 
  "167012^Nazjatar^^Brinestone Pickaxe", 
  "168081^Nazjatar^^Brinestone Pickaxe", 
  "167059^Nazjatar^^Chum", 
  "167060^Nazjatar^^Chum", 
  "168155^Nazjatar^^Chum", 
  "168159^Nazjatar^^Chum", 
  "167923^Nazjatar^^Clean Murloc Sock", 
  "169783^Nazjatar^^Cultist Pinky Finger", 
  "167905^Nazjatar^^Curious Murloc Horn", 
  "167916^Nazjatar^^Dirty Murloc Sock", 
  "167903^Nazjatar^^Disintegrating Sand Sculpture", 
  "170167^Nazjatar^Reputation^Eel Fillet", 
  "170472^Nazjatar^^Encrusted Coin", 
  "167907^Nazjatar^^Extra-Slimy Snail", 
  "168094^Nazjatar^^Faintly Humming Sea Stones", 
  "170176^Nazjatar^Reputation^Fathom Ray Wing", 
  "167906^Nazjatar^^Flatulent Fish", 
  "166888^Nazjatar^^Germinating Seed", 
  "167786^Nazjatar^^Germinating Seed", 
  "167909^Nazjatar^^Ghost Food", 
  "170171^Nazjatar^Reputation^Giant Crab Leg", 
  "167913^Nazjatar^^Healthy Murloc Lunch", 
  "170100^Nazjatar^^Hungry Herald's Tentacle Taco", 
  "167914^Nazjatar^^Jar of Fish Faces", 
  "167911^Nazjatar^^Just Regular Butter", 
  "170512^Nazjatar^^Lesser Benthic Arcanocrystal", 
  "170547^Nazjatar^^Mardivas's Arcane Cache Key", 
  "170477^Nazjatar^^Mardivas's Universally Lauded Tote", 
  "168161^Nazjatar^^Molted Shell", 
  "170085^Nazjatar^Reputation^Naga Deployment Orders", 
  "170153^Nazjatar^^Ominous Looking Tome", 
  "169781^Nazjatar^^Overwhelmingly-Alluring Idol", 
  "167902^Nazjatar^^Particularly Dense Rock", 
  "168097^Nazjatar^^Pilfered Armor Crate", 
  "167893^Nazjatar^^Prismatic Crystal", 
  "169780^Nazjatar^^Pulsating Blood Stone", 
  "170180^Nazjatar^^Razorshell", 
  "168261^Nazjatar^^Reinforced Cache Key", 
  "167077^Nazjatar^^Scrying Stone", 
  "167908^Nazjatar^^Sea Giant Foot Dust", 
  "170191^Nazjatar^^Skeletal Hand", 
  "167896^Nazjatar^^Slimy Naga Eyeball", 
  "167904^Nazjatar^^Smelly Pile of Gloop", 
  "169332^Nazjatar^^Strange Mineralized Water", 
  "169334^Nazjatar^^Strange Oceanic Sediment", 
  "169333^Nazjatar^^Strange Volcanic Rock", 
  "167915^Nazjatar^^Sweet Sea Vegetable", 
  "170181^Nazjatar^^Tidal Guard", 
  "167912^Nazjatar^^Unidentified Mass", 
  "170158^Nazjatar^^Unspeakable Pearl Idol", 
  "168053^Nazjatar^^Unusually Wise Hermit Crab", 
  "170161^Nazjatar^Reputation^Unusually Wise Hermit Crab", 
  "169942^Nazjatar^Reputation^Vibrant Sea Blossom", 
  "170162^Nazjatar^^Waterlogged Toolbox", 
  "170502^Nazjatar^^Waterlogged Toolbox", 
  "167832^Mechagon^Zone Specific Food^Canned Minnows", 
  "168232^Nazjatar^Zone Specific Food^Murloco's 'Fish' Tacos", 
  "170183^Nazjatar^Reputation^Reefwalker Bark", 
  "168666^Nazjatar^^Hefty Glimmershell", 
  "174325^Patch8_3^^Fire Bomb", 
  "172072^Patch8_3^^Experimental Vial", 
  "174351^Patch8_3^^K'Bab", 
  "174041^Patch8_3^^Eyeball Jelly", 
  "174348^Patch8_3^^Grilled Gnasher", 
  "174349^Patch8_3^^Ghastly Goulash", 
  "174352^Patch8_3^^Baked Voidfin", 
  "174350^Patch8_3^^Dubious Delight", 
  "174768^Patch8_3^^Cursed Relic", 
  "173937^Patch8_3^^Severed Oculus", 
  "174765^Patch8_3^^Tol'vir Relic", 
  "174761^Patch8_3^^Aqir Relic", 
  "174767^Patch8_3^^Mogu Relic", 
  "174766^Patch8_3^^Mantid Relic", 
  "174756^Patch8_3^^Aqir Relic Fragment", 
  "174858^Patch8_3^^Gersahl Greens", 
  "174758^Patch8_3^^Voidwarped Relic Fragment", 
  "174760^Patch8_3^^Mantid Relic Fragment", 
  "174764^Patch8_3^^Tol'vir Relic Fragment", 
  "168160^Patch8_3^^Jeweled Scarab Figurine", 
  "174020^Patch8_3^^N'lyeth, Sliver of N'Zoth", 
  "174759^Patch8_3^^Mogu Relic Fragment", 
  "174867^Patch8_3^^Shard of Corruption", 
  "174046^Patch8_3^^Orb of Visions", 
  "171334^Patch8_3^^Void-Touched Cloth", 
  "174045^Patch8_3^^Orb of Dark Portents", 
  "171347^Patch8_3^^Corrupted Bone Fragment", 
  "168267^Patch8_3^^Suntouched Figurine", 
  "168271^Patch8_3^^Stolen Ramkahen Banner", 
  "172947^Patch8_3^^Faceless Mask of Dark Imagination", 
  "171212^Patch8_3^^Sanity Restoration Orb", 
  "169294^Patch8_3^^Resilient Soul", 
  "173293^Patch8_3^^Vial of Self Preservation", 
  "172494^Patch8_3^^Baruk Idol", 
  "167788^Patch8_3^^Detoxifying Vial", 
  "173888^Patch8_3^^Shard of Self Sacrifice", 
  "175150^Patch8_3^^Self-Shaping Amber", 
  "167027^Patch8_3^^Portable Clarity Beam", 
  "173363^Patch8_3^Visions^Vessel of Horrific Visions", 
  "173375^Patch8_3^Reputation^Rajani Insignia", 
  "173376^Patch8_3^Reputation^Uldum Accord Insignia", 
  "173417^Patch8_3^Black^Black Empire Cloth Helm", 
  "173419^Patch8_3^Black^Black Empire Cloth Spaulders", 
  "173414^Patch8_3^Black^Black Empire Cloth Robes", 
  "173423^Patch8_3^Black^Black Empire Cloth Bracers", 
  "173416^Patch8_3^Black^Black Empire Cloth Gloves", 
  "173420^Patch8_3^Black^Black Empire Cloth Belt", 
  "173418^Patch8_3^Black^Black Empire Cloth Leggings", 
  "173415^Patch8_3^Black^Black Empire Cloth Boots", 
  "173410^Patch8_3^Black^Black Empire Leather Helm", 
  "173412^Patch8_3^Black^Black Empire Leather Spaulders", 
  "173407^Patch8_3^Black^Black Empire Leather Chestpiece", 
  "173424^Patch8_3^Black^Black Empire Leather Bracers", 
  "173409^Patch8_3^Black^Black Empire Leather Gloves", 
  "173413^Patch8_3^Black^Black Empire Leather Belt", 
  "173411^Patch8_3^Black^Black Empire Leather Leggings", 
  "173408^Patch8_3^Black^Black Empire Leather Boots", 
  "173403^Patch8_3^Black^Black Empire Mail Helm", 
  "173405^Patch8_3^Black^Black Empire Mail Spaulders", 
  "173400^Patch8_3^Black^Black Empire Mail Chestpiece", 
  "173425^Patch8_3^Black^Black Empire Mail Bracers", 
  "173406^Patch8_3^Black^Black Empire Mail Gloves", 
  "173402^Patch8_3^Black^Black Empire Mail Belt", 
  "173404^Patch8_3^Black^Black Empire Mail Leggings", 
  "173401^Patch8_3^Black^Black Empire Mail Boots", 
  "173396^Patch8_3^Black^Black Empire Plate Helm", 
  "173398^Patch8_3^Black^Black Empire Plate Spaulders", 
  "173393^Patch8_3^Black^Black Empire Plate Chestpiece", 
  "173422^Patch8_3^Black^Black Empire Plate Bracers", 
  "173399^Patch8_3^Black^Black Empire Plate Gloves", 
  "173395^Patch8_3^Black^Black Empire Plate Belt", 
  "173397^Patch8_3^Black^Black Empire Plate Leggings", 
  "173394^Patch8_3^Black^Black Empire Plate Boots" 
}


AdiBags_BfAZoneItems.lua
--[[
AdiBags_PriorExpansions - Seperates items from current expansion from those from prior ones, an addition to Adirelle's fantastic bag addon AdiBags.
Copyright 2019 Ggreg Taylor
All rights reserved.
--]]

local addon = LibStub('AceAddon-3.0'):GetAddon('AdiBags')
local L = setmetatable({}, {__index = addon.L})

local kCategory = 'Zone Item'
local kPfx = '|cff00ffff' 
local kSfx = '|r'
local Ggbug = false

local addonName2, addon2 = ...
local ZONE_ITEMS = addon2.ZONE_ITEMS 

local setFilter = addon:RegisterFilter("ZoneItems", 47, 'ABEvent-1.0')
setFilter.uiName = L['Zone Specific Items']
setFilter.uiDesc = L['Group zone specific items together.']

function setFilter:OnInitialize(b)
  self.db = addon.db:RegisterNamespace('ZoneItems', {
    profile = { enable = true },
    char = {  },
  })
end

function setFilter:Update()
  self:SendMessage('AdiBags_FiltersChanged')
end
function setFilter:OnEnable()
  addon:UpdateFilters()
end
function setFilter:OnDisable()
  addon:UpdateFilters()
end

local setNames = {}

function setFilter:GetOptions()
  return {
    enableZoneItem = {
      name = L['Enable Zone Item groups'],
      desc = L['Check this if you want to automatically seperate Nazjatar and Mechagon items.'],
      type = 'toggle',
      order = 25,
    },
    groupSetZoneItemSubgroups = {
      name = L['Further Zone Item Sub-Grouping Options'],
      type = L['group'],
      inline = true,
      order = 26,
      args = {
        _desc = {
          name = L['Select optional additional sub-groupings.'],
          type = 'description',
          order = 10,
        }, 
        groupBenthic = {
          name = L['Benthic BoA Items'],
          desc = L['Group Benthic Bind on Account gear tokens seperately.'],
          type = 'toggle',
          order = 27,
        },
        groupBlack = {
          name = L['Black Empire BoA Items'],
          desc = L['Group Black Empire Bind on Account gear tokens seperately.'],
          type = 'toggle',
          order = 28,
        },
        groupRepItems = {
          name = L['Reputation Items'],
          desc = L['Group Reputation on-use and repeatable turn-in items seperately.'],
          type = 'toggle',
          order = 29,
        },
        groupEssences = {
          name = L['Heart Essences'],
          desc = L['Group Heart of Azeroth essences seperately.'],
          type = 'toggle',
          order = 30,
        },
        groupPatch8_3 = {
          name = L['Patch 8.3 Items'],
          desc = L['Group items added in Patch 8.3 for Uldum, Horrific Visions, and Vale of Eternal Blossoms. They really should leave that poor Vale alone.'],
          type = 'toggle',
          order = 31,
        },
        groupVisions = {
          name = L['Visions'],
          desc = L['Group items for Visions.'],
          type = 'toggle',
          order = 32,
        },
        zonePriority = {
          name = L['Current Zone First'],
          desc = L['Group current zone\'s items first in bags.'],
          type = 'toggle',
          order = 33,
        },
      }
    },
  }, addon:GetOptionHandler(self, false, function() return self:Update() end)
end

-----------------------
function setFilter:Filter(slotData)
  local currZoneName = GetRealZoneText()
  if self.db.profile.groupEssences then
    addon:SetCategoryOrder('Essence',29)
  end
  if self.db.profile.groupRepItems then
    addon:SetCategoryOrder('Current Rep Item',30)
  end
  if self.db.profile.groupPatch8_3 then
    addon:SetCategoryOrder('Patch 8.3',31)
  end
  if self.db.profile.groupVisions then
    addon:SetCategoryOrder('Visions',32)
  end
  if self.db.profile.zonePriority then
    addon:SetCategoryOrder('Current Zone Item',33)
  end
    -- Exit if profile not enabled
  if (self.db.profile.enableZoneItem == false) or (slotData.itemId == false) then 
    return
  end
  local ziID, ziZone, ziSubcat, ziName, currSubCategory, bagItemID
  bagItemID = tonumber(slotData.itemId)
  -- Check if Heart of Azeroth Essence, and grouping for them selected
  local item = Item:CreateFromBagAndSlot(slotData.bag, slotData.slot)
  local _,_, itemRarity, _,_, itemType, itemSubType, _,_,_,_,_,_,_,_ = GetItemInfo(slotData.itemId)
  if (self.db.profile.groupEssences) and (itemType == 'Consumable' and itemSubType == 'Other' and itemRarity == 6) then
    return kPfx .. 'Heart Essence'.. kSfx, 'Essence'
  else
  -- load array category/subcat values
    for x = 1, #ZONE_ITEMS do
      local currSubset = {}
      local currZoneItem = ZONE_ITEMS[x]
      local index = 1
      for w in currZoneItem:gmatch('([^^]+)') do 
        currSubset[index]  = w 
        index = index +1
      end
      --169478^BfA^Benthic^Benthic Bracers (Exampleb)
      --167027^Patch8_3^^Portable Clarity Beam
      ziID = tonumber(currSubset[1])
      ziZone = currSubset[2]
      ziSubcat = currSubset[3]
      ziName = currSubset[4]
      if bagItemID == ziID then
        --check Benthic
        if ziZone == 'BfA' then
          if ziSubcat == 'Benthic' and  (self.db.profile.groupBenthic) then
            currSubCategory = 'Benthic'
            return kPfx .. currSubCategory.. kSfx, kCategory
          elseif ziSubcat == 'Essence' and (self.db.profile.groupEssences) then
            currSubCategory = 'Heart Essence'
            return kPfx .. currSubCategory .. kSfx, 'Essence'
          end
        elseif ziZone == 'Patch8_3' and (self.db.profile.groupPatch8_3) then
          if ziSubcat == 'Reputation' then
            currSubCategory = 'Reputation'
            return kPfx .. currSubCategory.. kSfx, 'Current Rep Item'
          elseif ziSubcat == 'Visions' and (self.db.profile.groupVisions) then
            currSubCategory = 'Visions'
            return kPfx .. currSubCategory.. kSfx, 'Visions'
          elseif ziSubcat == 'Black' and (self.db.profile.groupBlack) then
            currSubCategory = 'Black Empire'
            return kPfx .. currSubCategory .. kSfx, 'Black Empire'
          else
            currSubCategory = 'Patch 8.3 Item'
            return kPfx .. currSubCategory.. kSfx, 'Patch 8.3 Item'
          end
        elseif ziZone == 'Nazjatar'  or ziZone == 'Mechagon' then
          if (self.db.profile.groupRepItems) and ziSubcat=='Reputation' then
            currSubCategory = 'Reputation'
            return kPfx .. currSubCategory.. kSfx, 'Current Rep Item'
          end
          currSubCategory = ziZone
          if self.db.profile.zonePriority and ziZone == currZoneName then
            return kPfx .. currSubCategory.. kSfx, 'Current Zone Item'
          else
            return kPfx .. currSubCategory.. kSfx, kCategory
          end
        end
      end
    end
  end
end

